lambdas/link_fetcher/tests/test_subscription_endpoint.py

```diff
@@ -0,0 +1,289 @@
+        ) as mock_add_to_db_and_sqs:
+            process_notification(
+                event_s2_created,
+                {"none"},
+                lambda: db_session,
+                # ensure we granule is recent enough <30 days
+                now_utc=lambda: datetime.fromisoformat(
+                    event_s2_created["value"]["PublicationDate"]
+                ),
+            )
+        spy_filter_search_results.assert_called_once()
+        mock_add_to_db_and_sqs.assert_not_called()
+
+
+class TestApp:
+    """Test API endpoint"""
+
+    @pytest.fixture()
+    def config(self) -> EndpointConfig:
+        return EndpointConfig(
+            stage="local",
+            notification_username="test_user",
+            notification_password="password",
+        )
+
+    @pytest.fixture
+    def app(
+        self, config: EndpointConfig, db_connection_secret, mock_sqs_queue
+    ) -> FastAPI:
+        self.endpoint_config = config
+        self.db_connection_secret = db_connection_secret
+        self.mock_sqs_queue = mock_sqs_queue
+        return build_app(config)
+
+    @pytest.fixture
+    def test_client(self, app: FastAPI) -> TestClient:
+        return TestClient(app)
+
+    def test_handles_new_created_event(
+        self, test_client: TestClient, event_s2_created: dict
+    ):
+        """Test happy path for handling subscription event, mocking processing function"""
+        with patch(
+            "app.subscription_endpoint.process_notification", Mock()
+        ) as mock_process_notification:
+            resp = test_client.post(
+                "/events",
+                json=event_s2_created,
+                auth=(
+                    self.endpoint_config.notification_username,
+                    self.endpoint_config.notification_password,
+                ),
+            )
+        # processed successfully but no content
+        resp.raise_for_status()
+        assert resp.status_code == 204
+        mock_process_notification.assert_called_once()
+
+    def test_handles_new_created_event_is_added(
+        self,
+        test_client: TestClient,
+        db_session: Session,
+        event_s2_created: dict,
+    ):
+        """Test happy path for handling subscription event, mocking DB and SQS
+
+        This is also distinguished by using a "recent" event notification that works
+        by freezing/rewinding time to acquisition date for the test duration.
+        """
+        # we can't inject an older "utc_now" into `process_notification` so we have to
+        # patch `datetime.now()` with freezegun
+        with freeze_time(event_s2_created["value"]["PublicationDate"]):
+            resp = test_client.post(
+                "/events",
+                json=event_s2_created,
+                auth=(
+                    self.endpoint_config.notification_username,
+                    self.endpoint_config.notification_password,
+                ),
+            )
```

Comment:
Actually, I believe we _can_ inject an older "utc_now" by doing the following, along with the suggestion I made for `subscription_endpoint.py`.

However, feel free to skip this (along with the corresponding suggestion on `subscription_endpoint.py`), although I'm curious to see if this would work.

```suggestion
        setattr(
            test_client.app,
            "now_utc",
            lambda: datetime.fromisoformat(
                event_s2_created["value"]["PublicationDate"]
            ),
        )
        resp = test_client.post(
            "/events",
            json=event_s2_created,
            auth=(
                self.endpoint_config.notification_username,
                self.endpoint_config.notification_password,
            ),
        )
```